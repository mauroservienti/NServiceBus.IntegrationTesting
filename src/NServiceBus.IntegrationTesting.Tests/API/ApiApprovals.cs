using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using PublicApiGenerator;

namespace NServiceBus.IntegrationTesting.Tests.API
{
    public class APIApprovals
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve_API()
        {
            var publicApi = typeof(IntegrationScenarioContext).Assembly.GeneratePublicApi();
            Approvals.Verify(publicApi);
        }
    }
}