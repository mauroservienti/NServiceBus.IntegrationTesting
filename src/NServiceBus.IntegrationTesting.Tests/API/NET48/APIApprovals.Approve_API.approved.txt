[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("NServiceBus.IntegrationTesting.Tests")]
namespace NServiceBus.IntegrationTesting
{
    public class EmptyTestCompletionHandler : NServiceBus.IntegrationTesting.IHandleTestCompletion
    {
        public EmptyTestCompletionHandler() { }
        public System.Threading.Tasks.Task OnTestCompleted(NServiceBus.AcceptanceTesting.Support.RunSummary summary) { }
    }
    public static class EndpointConfigurationExtensions
    {
        public static void RegisterRequiredPipelineBehaviors(this NServiceBus.EndpointConfiguration builder, string endpointName, NServiceBus.IntegrationTesting.IntegrationScenarioContext integrationScenarioContext) { }
        public static void RegisterScenarioContext(this NServiceBus.EndpointConfiguration builder, NServiceBus.AcceptanceTesting.ScenarioContext scenarioContext) { }
    }
    public class EndpointReference
    {
        public static NServiceBus.IntegrationTesting.EndpointReference FromSolutionProject(string projectName) { }
    }
    public abstract class EndpointTemplate : NServiceBus.AcceptanceTesting.Support.IEndpointSetupTemplate
    {
        protected EndpointTemplate() { }
        public System.Threading.Tasks.Task<NServiceBus.EndpointConfiguration> GetConfiguration(NServiceBus.AcceptanceTesting.Support.RunDescriptor runDescriptor, NServiceBus.AcceptanceTesting.Support.EndpointCustomizationConfiguration endpointCustomizationConfiguration, System.Action<NServiceBus.EndpointConfiguration> configurationBuilderCustomization) { }
        protected abstract System.Threading.Tasks.Task<NServiceBus.EndpointConfiguration> OnGetConfiguration(NServiceBus.AcceptanceTesting.Support.RunDescriptor runDescriptor, NServiceBus.AcceptanceTesting.Support.EndpointCustomizationConfiguration endpointCustomizationConfiguration, System.Action<NServiceBus.EndpointConfiguration> configurationBuilderCustomization);
    }
    public class EndpointTemplate<T> : NServiceBus.IntegrationTesting.EndpointTemplate
        where T : NServiceBus.EndpointConfiguration, new ()
    {
        public EndpointTemplate() { }
        protected override System.Threading.Tasks.Task<NServiceBus.EndpointConfiguration> OnGetConfiguration(NServiceBus.AcceptanceTesting.Support.RunDescriptor runDescriptor, NServiceBus.AcceptanceTesting.Support.EndpointCustomizationConfiguration endpointCustomizationConfiguration, System.Action<NServiceBus.EndpointConfiguration> configurationBuilderCustomization) { }
    }
    public class EndpointTemplate<T, C> : NServiceBus.IntegrationTesting.EndpointTemplate<T>
        where T : NServiceBus.EndpointConfiguration, new ()
        where C : NServiceBus.IntegrationTesting.IHandleTestCompletion, new ()
    {
        public EndpointTemplate() { }
        protected override System.Threading.Tasks.Task<NServiceBus.EndpointConfiguration> OnGetConfiguration(NServiceBus.AcceptanceTesting.Support.RunDescriptor runDescriptor, NServiceBus.AcceptanceTesting.Support.EndpointCustomizationConfiguration endpointConfiguration, System.Action<NServiceBus.EndpointConfiguration> configurationBuilderCustomization) { }
    }
    public class GenericHostEndpointBehaviorBuilder<TContext>
        where TContext : NServiceBus.AcceptanceTesting.ScenarioContext
    {
        public GenericHostEndpointBehaviorBuilder() { }
        public System.Collections.Generic.IList<NServiceBus.AcceptanceTesting.Support.IWhenDefinition> Whens { get; }
        public NServiceBus.IntegrationTesting.GenericHostEndpointBehaviorBuilder<TContext> When(System.Func<NServiceBus.IMessageSession, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.GenericHostEndpointBehaviorBuilder<TContext> When(System.Func<NServiceBus.IMessageSession, TContext, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.GenericHostEndpointBehaviorBuilder<TContext> When(System.Func<TContext, System.Threading.Tasks.Task<bool>> condition, System.Func<NServiceBus.IMessageSession, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.GenericHostEndpointBehaviorBuilder<TContext> When(System.Func<TContext, System.Threading.Tasks.Task<bool>> condition, System.Func<NServiceBus.IMessageSession, TContext, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.GenericHostEndpointBehaviorBuilder<TContext> When(System.Predicate<TContext> condition, System.Func<NServiceBus.IMessageSession, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.GenericHostEndpointBehaviorBuilder<TContext> When(System.Predicate<TContext> condition, System.Func<NServiceBus.IMessageSession, TContext, System.Threading.Tasks.Task> action) { }
    }
    public interface IHandleTestCompletion
    {
        System.Threading.Tasks.Task OnTestCompleted(NServiceBus.AcceptanceTesting.Support.RunSummary summary);
    }
    public interface IRemoteEndpointWhenDefinition
    {
        System.Guid Id { get; }
        System.Threading.Tasks.Task<bool> ExecuteAction(NServiceBus.IntegrationTesting.IntegrationScenarioContext context, NServiceBus.IntegrationTesting.IRemoteMessageSessionProxy session);
    }
    public class IntegrationScenarioContext : NServiceBus.AcceptanceTesting.ScenarioContext
    {
        public IntegrationScenarioContext() { }
        public System.Collections.Generic.IEnumerable<NServiceBus.IntegrationTesting.HandlerInvocation> InvokedHandlers { get; }
        public System.Collections.Generic.IEnumerable<NServiceBus.IntegrationTesting.SagaInvocation> InvokedSagas { get; }
        public System.Collections.Generic.IEnumerable<NServiceBus.IntegrationTesting.OutgoingMessageOperation> OutgoingMessageOperations { get; }
        public System.Collections.Generic.IReadOnlyDictionary<string, string> GetProperties(string endpointName) { }
        public bool HandlerWasInvoked<THandler>() { }
        public bool HasFailedMessages() { }
        public bool HasHandlingErrors() { }
        public bool MessageWasProcessed<TMessage>() { }
        public bool MessageWasProcessedByHandler<TMessage, THandler>() { }
        public bool MessageWasProcessedBySaga<TMessage, TSaga>() { }
        public void RegisterTimeoutRescheduleRule<TTimeout>(System.Func<object, NServiceBus.DelayedDelivery.DoNotDeliverBefore, NServiceBus.DelayedDelivery.DoNotDeliverBefore> rule) { }
        public bool SagaWasCompleted<TSaga>()
            where TSaga : NServiceBus.Saga { }
        public bool SagaWasInvoked<TSaga>()
            where TSaga : NServiceBus.Saga { }
    }
    public class OutOfProcessEndpointBehaviorBuilder<TContext>
        where TContext : NServiceBus.IntegrationTesting.IntegrationScenarioContext
    {
        public OutOfProcessEndpointBehaviorBuilder() { }
        public System.Collections.Generic.IList<NServiceBus.IntegrationTesting.IRemoteEndpointWhenDefinition> Whens { get; }
        public NServiceBus.IntegrationTesting.OutOfProcessEndpointBehaviorBuilder<TContext> When(System.Func<NServiceBus.IntegrationTesting.IRemoteMessageSessionProxy, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.OutOfProcessEndpointBehaviorBuilder<TContext> When(System.Func<NServiceBus.IntegrationTesting.IRemoteMessageSessionProxy, TContext, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.OutOfProcessEndpointBehaviorBuilder<TContext> When(System.Func<TContext, System.Threading.Tasks.Task<bool>> condition, System.Func<NServiceBus.IntegrationTesting.IRemoteMessageSessionProxy, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.OutOfProcessEndpointBehaviorBuilder<TContext> When(System.Func<TContext, System.Threading.Tasks.Task<bool>> condition, System.Func<NServiceBus.IntegrationTesting.IRemoteMessageSessionProxy, TContext, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.OutOfProcessEndpointBehaviorBuilder<TContext> When(System.Predicate<TContext> condition, System.Func<NServiceBus.IntegrationTesting.IRemoteMessageSessionProxy, System.Threading.Tasks.Task> action) { }
        public NServiceBus.IntegrationTesting.OutOfProcessEndpointBehaviorBuilder<TContext> When(System.Predicate<TContext> condition, System.Func<NServiceBus.IntegrationTesting.IRemoteMessageSessionProxy, TContext, System.Threading.Tasks.Task> action) { }
    }
    public static class ScenarioWithEndpointBehaviorExtensions
    {
        public static NServiceBus.AcceptanceTesting.Support.IScenarioWithEndpointBehavior<TContext> WithGenericHostEndpoint<TContext>(this NServiceBus.AcceptanceTesting.Support.IScenarioWithEndpointBehavior<TContext> scenarioWithEndpoint, string endpointName, System.Func<System.Action<NServiceBus.EndpointConfiguration>, Microsoft.Extensions.Hosting.IHost> hostBuilder, System.Action<NServiceBus.IntegrationTesting.GenericHostEndpointBehaviorBuilder<TContext>> behavior = null)
            where TContext : NServiceBus.AcceptanceTesting.ScenarioContext { }
        public static NServiceBus.AcceptanceTesting.Support.IScenarioWithEndpointBehavior<TContext> WithOutOfProcessEndpoint<TContext>(this NServiceBus.AcceptanceTesting.Support.IScenarioWithEndpointBehavior<TContext> scenarioWithEndpoint, string endpointName, NServiceBus.IntegrationTesting.EndpointReference endpointReference, System.Action<NServiceBus.IntegrationTesting.OutOfProcessEndpointBehaviorBuilder<TContext>> behavior = null)
            where TContext : NServiceBus.IntegrationTesting.IntegrationScenarioContext { }
    }
}